### PR TITLE
feat: add support to inject script to our backends init script, new var custom_data

### PR DESCRIPTION
**Pull request by denise89p**

### CODE REVIEW

File: prerequisites.tf [hunk 1]
```diff
@@ -39,16 +39,18 @@ module "iam" {
   secret_prefix                   = local.secret_prefix
   tiering_enable_obs_integration  = var.tiering_enable_obs_integration
   additional_iam_policy_statement = var.additional_instance_iam_policy_statement
+  enable_ssm_policy               = var.vpc_endpoint_ssm_create
 }
 
 module "vpc_endpoint" {
-  count                          = var.vpc_endpoint_ec2_create || var.vpc_endpoint_proxy_create || var.vpc_endpoint_s3_gateway_create ? 1 : 0
+  count                          = var.vpc_endpoint_ssm_create || var.vpc_endpoint_ec2_create || var.vpc_endpoint_proxy_create || var.vpc_endpoint_s3_gateway_create ? 1 : 0
```
Comment (assafgi) [inline prerequisites.tf:46]: need to also add `var.vpc_endpoint_lambda_create`

File: modules/iam/variables.tf [hunk 2]
```diff
@@ -44,3 +44,9 @@ variable "additional_iam_policy_statement" {
     error_message = "Additional IAM policy statement must be a non-empty list (if provided)."
   }
 }
+
+variable "enable_ssm_policy" {
+  type        = bool
+  default     = false
+  description = "Attache AmazonSSMManagedEC2InstanceDefaultPolicy policy to allow SSM"
```
Comment (assafgi) [inline modules/iam/variables.tf:51]: typo: `Attach`

Comment (denise89p) [reply modules/iam/variables.tf]: removed

File: prerequisites.tf (hunk 1 above)
Comment (denise89p) [reply prerequisites.tf]: removed

**Review by assafgi — APPROVED** [review]

**Pull request merged**